rust/connlib/shared/src/callbacks.rs

```diff
@@ -8,6 +9,36 @@ use std::path::PathBuf;
 // Avoids having to map types for Windows
 type RawFd = i32;
 
+#[derive(Serialize, Clone, Copy, Debug)]
+pub struct Cidrv4 {
+    address: Ipv4Addr,
+    prefix: u8,
+}
+
+#[derive(Serialize, Clone, Copy, Debug)]
```

Comment:
```suggestion
/// Identical to `ip_network::Ipv6Network` except we implement `Serialize` on the Rust side and the equivalent of `Deserialize` on the Swift / Kotlin side to avoid manually serializing and deserializing.
#[derive(Serialize, Clone, Copy, Debug)]
```